libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2416,79 +2289,66 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
-
-		auto [errorId, description] = [&]() -> std::tuple<ErrorId, std::string> {
-			std::string msg = isVariadic ?
-				"Need at least " +
-				toString(parameterTypes.size()) +
-				" arguments for " +
-				std::string(isStructConstructorCall ? "struct constructor" : "function call") +
-				", but provided only " +
-				toString(arguments.size()) +
-				"."
-				:
-				"Wrong argument count for " +
-				std::string(isStructConstructorCall ? "struct constructor" : "function call") +
-				": " +
-				toString(arguments.size()) +
-				" arguments given but " +
-				std::string(isVariadic ? "need at least " : "expected ") +
-				toString(parameterTypes.size()) +
-				".";
+		bool const isStructConstructorCall = functionCallKind == FunctionCallKind::StructConstructorCall;
+
+		auto [errorId, description] = [&]() -> std::tuple<ErrorId, std::string>
+		{
+			std::string msg = isVariadic
+								  ? "Need at least " + toString(parameterTypes.size()) + " arguments for "
+										+ std::string(isStructConstructorCall ? "struct constructor" : "function call")
+										+ ", but provided only " + toString(arguments.size()) + "."
+								  : "Wrong argument count for "
+										+ std::string(isStructConstructorCall ? "struct constructor" : "function call")
+										+ ": " + toString(arguments.size()) + " arguments given but "
+										+ std::string(isVariadic ? "need at least " : "expected ")
+										+ toString(parameterTypes.size()) + ".";
 
 			if (isStructConstructorCall)
 			{
 				solAssert(!isVariadic, "");
-				return { 9755_error, msg };
+				return {9755_error, msg};
 			}
 			else if (
-				_functionType->kind() == FunctionType::Kind::BareCall ||
-				_functionType->kind() == FunctionType::Kind::BareCallCode ||
-				_functionType->kind() == FunctionType::Kind::BareDelegateCall ||
-				_functionType->kind() == FunctionType::Kind::BareStaticCall
-			)
+				_functionType->kind() == FunctionType::Kind::BareCall
+				|| _functionType->kind() == FunctionType::Kind::BareCallCode
+				|| _functionType->kind() == FunctionType::Kind::BareDelegateCall
+				|| _functionType->kind() == FunctionType::Kind::BareStaticCall)
 			{
 				solAssert(!isVariadic, "");
 				if (arguments.empty())
 					return {
 						6138_error,
-						msg +
-						" This function requires a single bytes argument."
-						" Use \"\" as argument to provide empty calldata."
+						msg
+							+ " This function requires a single bytes argument."
+							  " Use \"\" as argument to provide empty calldata."
 					};
 				else
 					return {
 						8922_error,
-						msg +
-						" This function requires a single bytes argument."
-						" If all your arguments are value types, you can use"
-						" abi.encode(...) to properly generate it."
+						msg
+							+ " This function requires a single bytes argument."
+							  " If all your arguments are value types, you can use"
+							  " abi.encode(...) to properly generate it."
```

Comment:
I'm ambivalent about where to put the operator (end of previous line or start of the new one), but the thing I don't like here is, again, the fact that putting it on a new line seems to introduce alignment with spaces.

Maybe keeping operators at the end would prevent this?